welcome/welcome.py

```diff
@@ -778,10 +778,15 @@ def __init__(self, *args, **kwargs) -> None:
             plural = "s"
 
         channel = await self.__get_channel(guild, event)
+        if isinstance(user, discord.Member):
+            roles = [r.name for r in user.roles if r.name != "@everyone"]
+            roles = self.format_list(*roles)
+        else:
+            roles = []
 
         try:
             return await channel.send(
-                format_str.format(member=user, server=guild, bot=user, count=count or "", plural=plural)
+                format_str.format(member=user, server=guild, bot=user, count=count or "", plural=plural, roles=roles)
```

Comment:
Make sure this is documented in all four places where the user can add new message formats (one for each event). I recommend this:
> \`{roles}\` is a list of all the roles that the member has at the time

Reply:
done